monitor/dashboard.py

```diff
@@ -85,6 +93,37 @@ def verify_cached_stats(self, cached_stats):
             data = json.loads(cached_stats)
         return data
 
+    def add_supply_endpoint(self, flask_server: Flask, economics: BaseEconomics):
+        @flask_server.route('/supply_information', methods=["GET"])
+        def supply_information():
+            supply_information = dict()
+
+            total_supply = economics.total_supply
+            initial_supply = economics.initial_supply
+
+            # Overview
+            supply_information['total_supply'] = str(round(NU.from_nunits(total_supply), 2))
+            supply_information['initial_supply'] = str(round(NU.from_nunits(initial_supply), 2))
+            supply_information['staking_rewards_supply'] = str(round(NU.from_nunits(total_supply - initial_supply), 2))
+
+            # Allocations
+            investor_supply = (0.319 + 0.08) * initial_supply  # SAFT1, SAFT2
+            team_supply = 0.106 * initial_supply  # Team
+            nuco_supply = 0.2 * initial_supply  # NuCo
+
+            initial_allocation = dict()
+            supply_information['internal_allocations'] = initial_allocation
+            initial_allocation['investors'] = str(round(NU.from_nunits(investor_supply), 2))
+            initial_allocation['team'] = str(round(NU.from_nunits(team_supply), 2))
+            initial_allocation['company'] = str(round(NU.from_nunits(nuco_supply), 2))
+
+            # Non-internal supply
+            est_circulating_supply = str(round(NU.from_nunits(initial_supply - investor_supply - team_supply - nuco_supply), 2))
```

Comment:
Clarification from CMC on their definition:

"""
Circulating supply is what is available to the public. This excludes any team/developer controlled assets, privately allocated assets, and assets allocated for any future use/distribution. We typically only exclude staked tokens that are controlled by the team.
"""

Reply:
So I think this should be Initial supply (+ issued rewards) - locked NU (saft 2, team, worklock, NuCo).

The tricky thing will be to either manually or automatically update the number as saft 2/team/worklock vest

Reply:
we could add a vesting calculation...similar to what is shown in the `Genesis Block Unlock Schedule` graph  in https://blog.nucypher.com/the-worklock/ - presumably relative to Oct 15th 00:00:00 UTC...?

Reply:
Yeah, relative to Oct 15 is a good approximation, although not everyone will have started their nodes then